src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,46 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userFromDb = userService.findByEmail(email);
+        if (userFromDb.isEmpty()) {
+            throw new AuthenticationException("Can't authenticate user: " + email);
+        }
+        User user = userFromDb.get();
+        String hashPassword = HashUtil.hashPassword(password, user.getSalt());
+        if (user.getPassword().equals(hashPassword)) {
+            return user;
+        }
+        throw new AuthenticationException("Can't authenticate user: " + email);
+    }
```

Comment:
```suggestion
        Optional<User> userFromDb = userService.findByEmail(email);
        if (userFromDb.isPresent() && checkPassword(userFromDb.get(), password)) {
           return userFromDb.get();
        }
        throw new AuthenticationException("Can't authenticate user: " + email);
    }
    
  private boolean checkPassword(User user, String password) {
       String hashedPassword = HashUtil.hashPassword(password, user.getSalt());
       return user.getPasswors().equals(hashedPassword);
  }
```

Reply:
It's optional advice, but if you follow checkstyle -> we have some advices there about it. Likt: 
"Create only one condition for throwing AuthenticationException in login() method. You may combine two checks: whether the user has been found by login and does passwords match."